W-12293473-linking-user-accounts (#285)
mulesoft/docs-access-management: modules/ROOT/pages/_partials/link-anypoint-platform-accounts.adoc

// end::profileLinkingMultiple[]

// tag::profileLinkingContextForAdmins[]
[[enable-profile-linking]]
= Enabling Users to Link Anypoint Platform Profiles

As an organization administrator, you can enable users to link multiple Anypoint Platform profiles that use the same email address to log in. 

Anypoint Platform supports:

* Linking an SSO login to a local profile by using credentials (login managed by Anypoint IdP)
* Linking an SSO login to another SSO profile that uses a different IdP
* Linking an SSO login to multiple other profiles that use different IdPs

Anypoint Platform does not support linking multiple profiles that use the same IdP. Note that only OpenID Connect (OIDC) IdPs are supported.

include::partial$include-nav-steps-org-admin.adoc[]

include::partial$include-nav-identity-providers.adoc[]

. In the *Link Multiple Accounts* section, click *Enabled*. 
. Click *Save Changes*. 
+
Users in your organization can now link their SSO logins with other IdP profiles that share the same email address.

// end::profileLinkingContextForAdmins[]

// tag::requireProfileLinkingForAdmins[]

> 

Author: ```suggestion

= Require Users to Link Anypoint Platform Profiles
```

// tag::profileLinkingContextForUsers[]
[[profile-linking-context-users]]
= Linking Your Anypoint Platform Profiles

When you log in to an Anypoint Platform using an email address that is associated with more than one profile in the same organization, you are prompted to link your profiles together. Your options for linking profiles vary based on whether the profiles use the Anypoint identity provider (IdP) or an external IdP that uses single sign-on (SSO). Anypoint Platform supports:

* Linking an SSO profile to a local profile (login managed by Anypoint IdP)
* Linking an SSO profile to another SSO profile that uses a different IdP to log in
* Linking an SSO profile to multiple other profiles that each use different IdPs to log in

Anypoint Platform does not support linking multiple profiles that use the same IdP. You can link up to 25 user profiles.

// end::profileLinkingContextForUsers[]

// tag::profileLinkingForSSOtoLocal[]
[[sso-to-local]]

> = Linking an SSO Profile to a Local Profile

Author: ```suggestion
= Linking an SSO Login to a Local Profile
```

> When you attempt to log in using SSO and there is another user that shares the same email address but uses Anypoint Platform credentials to log in, you are prompted to link the profiles.

Author: ```suggestion
When you attempt to log in using SSO and there is another user profile that shares the same email address but uses Anypoint Platform credentials to log in, you are prompted to link the profiles.
```

. In the Anypoint Platform login page, click *Sign in with <your identity provider>*.
. If prompted, enter your credentials for your IdP. +

> If your SSO credentials share an email address with a profile that is managed by the Anypoint IdP, you are prompted to log in. If you are not prompted, your local profile uses a different email address than your SSO login.

Reviewer: I think we can skip "If you are not prompted, your local profile uses a different email address than your SSO login." Using "your local profile" can be confusing because at the time of login, we/the service doesn't really know whether the customer/end user has an account or not. So let's not imply they do.

Author: ```suggestion
If your SSO credentials share an email address with a profile that is managed by the Anypoint IdP, you are prompted to log in.
```

> If your SSO credentials share an email address with an profile that is managed by the Anypoint IdP, you are prompted to log in. If you are not prompted, your local profile uses a different email address than your SSO profile.

Author: ```suggestion
If your SSO credentials share an email address with a profile that is managed by the Anypoint IdP, you are prompted to log in. If you are not prompted, your local profile uses a different email address than your SSO login.
```

+
. In the *Link your Account* page, enter your credentials for your local Anypoint Platform profile.
. Click *Link Accounts*.
+

> You can now view your linked profiles in the *Identity Providers* section of your user profile.

Reviewer: We just renamed "Identity Provider" to "Logged in via". 

Can we use singular "linked profile" as there can be only one in "Logged in via"? Just in case it causes confusion.

Author: ```suggestion
You can now view your linked profile in the *Logged in via* section of your user profile.
```

// end::profileLinkingForSSOtoLocal[]

// tag::profileLinkingSSOtoSSO[]
[[sso-to-sso]]

> = Linking an Existing SSO Login to Another SSO Login

Reviewer: High level comment about the structure of this page. Currently we have 3 sections for different linking scenarios, which is good, but, this may imply that the end user is able to initiate one of the 3 cases or they are able to choose. In fact, the end user doesn't have the ability to choose which cases they would step into. Email matching can be a black box to end users unless they are super clear about their emails and accounts in Anypoint. 

With this in mind, is it possible to unify the 3 cases in to just one workflow, and talk about potential options for an end user. A sample workflow is: (just an example)
1. Do sso login. 
2. You may be prompted with a 'Link an account' for you choose which existing account you want to login as and the selected user account will be linked to the sso login. 
3. After selecting a user, you may be promoted to authenticate using existing credentials linked/attached to this user account 
4. ..

Author: Sure. We don't have to include all of them. They're sectioned off, as this page isn't published as-is.

> When you attempt to log in using SSO and there is another user that shares the same email address but uses a different external identity provider with SSO, you are prompted to link the profiles.

Author: ```suggestion
When you attempt to log in using SSO and there is another user profile that shares the same email address but uses a different external identity provider with SSO, you are prompted to link the profiles.
```

. In the Anypoint Platform login page, click *Sign in with <your identity provider>*.
. If prompted, enter your credentials for your IdP. +

> If your SSO credentials share an email address with another SSO profile in the same organization, you are prompted to log in with those credentials to link your profiles. If you are not prompted, your other SSO profile uses a different email address than the SSO profile you are logging in to.

Author: ```suggestion
If your SSO credentials share an email address with another SSO login in the same organization, you are prompted to log in with those credentials to link your profiles. If you are not prompted, your other SSO login uses a different email address than the SSO login you are logging in to.
```

+
. In the *Link your Account* page, enter your credentials for your local Anypoint Platform profile.
. Click *Link Accounts*.
+
You can now view your linked profiles in the *Identity Providers* section of your user profile.

// end::profileLinkingSSOtoSSO[]

// tag::profileLinkingMultiple[]
[[linking-multiple-profiles]]

> = Linking an existing SSO Profile to Multiple Profiles

Author: ```suggestion
= Linking an existing SSO Login to Multiple Profiles
```

When you attempt to login using SSO and there are multiple user profiles that share the same email address but use different identity providers to log in, you are prompted to link the profiles. Note that you cannot link profiles that use the same IdP.

. In the Anypoint Platform login page, click *Sign in with <your identity provider>*.
. If prompted, enter your credentials for your IdP. +

> If your SSO credentials share an email address with additional profiles in the same organization, you are prompted to select one of the profiles and log in with those credentials to link your profiles. If you are not prompted, your other profiles use a different email address than the SSO profile you are logging in to.

Author: ```suggestion
If your SSO credentials share an email address with additional profiles in the same organization, you are prompted to select one of the profiles and log in with those credentials to link your profiles. If you are not prompted, your other profiles use a different email address than the SSO login you are logging in to.
```

+
. In the *Link your Account* page, select the profile you want to link. +
You can hover over each profile to see which identity provider it uses.
+
. Click *Link Accounts*.
. Enter your credentials for the profile you chose.
+
You can now view your linked profiles in the *Identity Providers* section of your user profile.

// end::profileLinkingMultiple[]

// tag::profileLinkingContextForAdmins[]
[[enable-profile-linking]]
= Enabling Users to Link Anypoint Platform Profiles

As an organization administrator, you can enable users to link multiple Anypoint Platform profiles that use the same email address to log in. 

Anypoint Platform supports:

* Linking an SSO login to a local profile by using credentials (login managed by Anypoint IdP)

> * Linking an SSO login to another SSO profile that uses a different IdP

Author: ```suggestion
* Linking an SSO login to another SSO login that uses a different IdP
```

* Linking an SSO login to multiple other profiles that use different IdPs

Anypoint Platform does not support linking multiple profiles that use the same IdP. Note that only OpenID Connect (OIDC) IdPs are supported.

include::partial$include-nav-steps-org-admin.adoc[]

include::partial$include-nav-identity-providers.adoc[]

. In the *Link Multiple Accounts* section, click *Enabled*. 
. Click *Save Changes*. 
+
Users in your organization can now link their SSO logins with other IdP profiles that share the same email address.

// end::profileLinkingContextForAdmins[]

// tag::requireProfileLinkingForAdmins[]

= Require Users to Link Anypoint Platform Profiles
// DO NOT USE THIS SECTION YET - NOT YET SUPPORTED 

If your organization has xref:external-identity-index.adoc#enable-profile-linking[enabled the *Link Multiple Accounts* setting], organization administrators can require all users to link available Anypoint Platform profiles.

> 

Reviewer: Shall we also include how to unlink? 

```suggestion
= Remove Link From User Account 
To remove the link from a user account, the Organization Admin use a DELETE API call to `/accounts/api/organizations/:orgId/users/:userId/identityProviderProfiles`. For details, refer to Access Management API references. 
```

Author: Who is able to do this? Can users do it to their own accounts?

Author: Also need a link to that "Access Management API references" doc, whatever that may be.

Reviewer: API docs: https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/access-management-api/

Organization admins can do it, i.e. as long as the user has an admin permission (Organization Administrator) they can do it. Regular users cannot

// tag::profileLinkingContextForUsers[]
[[profile-linking-context-users]]
= Linking Your Anypoint Platform Profiles

> When you log in to Anypoint Platform for the first time using an external identity provider (IdP), Anypoint Platform locates potential user accounts matched by your email address that is associated with the IdP profile. You are prompted to link your IdP profile with the existing accounts. Your options for linking profiles vary based on whether the profiles use the Anypoint identity provider (IdP) or an external IdP that uses single sign-on (SSO). Anypoint Platform supports:

Reviewer 2: Do you need the second sentence?

Author: ```suggestion
When you log in to Anypoint Platform for the first time using an external identity provider (IdP), Anypoint Platform locates potential user accounts matched by your email address that is associated with the IdP profile. Your options for linking profiles vary based on whether the profiles use the Anypoint identity provider (IdP) or an external IdP that uses single sign-on (SSO). Anypoint Platform supports:
```

> When you log in to Anypoint Platform for the first time using an external identity provider (IdP), Anypoint Platform locates potential user accounts matched by your email address that is associated with the IdP profile. You are prompted to link your IdP profile with the existing accounts. Your options for linking profiles vary based on whether the profiles use the Anypoint identity provider (IdP) or an external IdP that uses single sign-on (SSO). Anypoint Platform supports:

Reviewer 2: ```suggestion
When you log in to Anypoint Platform for the first time using an external identity provider (IdP), Anypoint Platform locates potential user accounts that match the email address associated with your IdP profile. Your options for linking profiles vary based on whether the profiles use the Anypoint identity provider (IdP) or an external IdP that uses single sign-on (SSO). Anypoint Platform supports:
```

* Linking an SSO login to a local profile by using credentials (login managed by Anypoint IdP)
* Linking an SSO login to another SSO login that uses a different IdP to log in
* Linking an SSO login to multiple other profiles that each use different IdPs to log in

Anypoint Platform does not support linking multiple profiles that use the same IdP. The IdP must be an OpenID Connect (OIDC) provider. You can link up to 25 user profiles. This feature is available only if an xref:external-identity-index.adoc#enable-profile-linking[organization administrator has enabled it].

// end::profileLinkingContextForUsers[]

// tag::profileLinking[]
[[profile-linking]]
= Linking an existing SSO Login to Additional Logins

> When you attempt to login using SSO and there are additional user profiles that share the same email address but use different identity providers (IdPs) to log in, you are prompted to link the profiles. Note that you cannot link profiles that use the same IdP.

Reviewer 2: ```suggestion
When you attempt to log in using SSO and there are additional user profiles that share the same email address but use different identity providers (IdPs) to log in, you are prompted to link the profiles. You cannot link profiles that use the same IdP.
```

. In the Anypoint Platform login page, click *Sign in with <your identity provider>*.
. If prompted, enter your credentials for your IdP. +

> If your SSO credentials share an email address with additional profiles in the same organization, you are prompted to select one of the profiles and log in with those credentials to link your profiles. If you are not prompted, your other profiles use a different email address than the SSO login you are logging in to.

Reviewer 2: ```suggestion
If your SSO credentials share an email address with additional profiles in the same organization, you are prompted to select one of the profiles and log in with those credentials to link your profiles. If you are not prompted, your other profiles use a different email address than the SSO login that you are logging in to.
```

Reviewer 2: Do you need the last sentence? I think its meaning is implied, but up to you.

Author: ```suggestion
If your SSO credentials share an email address with additional profiles in the same organization, you are prompted to select one of the profiles and log in with those credentials to link your profiles. 
```

+
. In the *Link your Account* page, select the profile you want to link. +
You can hover over each profile to see which identity provider it uses.
+
. Click *Link Accounts*.
. Enter your credentials for the profile you chose.
+
You can now view your linked profile in the *Logged In Via* section of your user profile.

// end::profileLinking[]

// tag::profileLinkingContextForAdmins[]
[[enable-profile-linking]]
= Enabling Users to Link Anypoint Platform Profiles

As an organization administrator, you can enable users to link multiple Anypoint Platform profiles that use the same email address to log in. 

Anypoint Platform supports:

* Linking an SSO login to a local profile by using credentials (login managed by Anypoint IdP)
* Linking an SSO login to another SSO login that uses a different IdP
* Linking an SSO login to multiple other profiles that use different IdPs

> Anypoint Platform does not support linking multiple profiles that use the same IdP. Note that only OpenID Connect (OIDC) IdPs are supported.

Reviewer 2: ```suggestion
Anypoint Platform does not support linking multiple profiles that use the same IdP. Only OpenID Connect (OIDC) IdPs are supported.
```

include::partial$include-nav-steps-org-admin.adoc[]

include::partial$include-nav-identity-providers.adoc[]

. In the *Link Multiple Accounts* section, click *Enabled*. 
. Click *Save Changes*. 
+
Users in your organization can now link their SSO logins with other IdP profiles that share the same email address.

// end::profileLinkingContextForAdmins[]

// tag::deleteLinkedProfilesAdmin[]
[[remove-linked-profile-org-admin]]

> = Remove Link From User Account 

Reviewer 2: Remove what link?
Remove a Link to...from a User Account

Author: ```suggestion
= Remove a Linked Profile From a User Account 
```

> Organization administrators can remove a linked profile from a user account by making a `DELETE` call to `/accounts/api/organizations/:orgId/users/:userId/identityProviderProfiles`. For details, refer to Access Management API references. 

Author: ```suggestion
Organization administrators can remove a linked profile from a user account by making a `DELETE` call to `/accounts/api/organizations/:orgId/users/:userId/identityProviderProfiles`. For details, refer to the https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/access-management-api/[Access Management API documentation]. 
```

> Organization administrators can remove a linked profile from a user account by making a `DELETE` call to `/accounts/api/organizations/:orgId/users/:userId/identityProviderProfiles`. For details, refer to Access Management API references. 

Reviewer 2: It is not clear to me what "Access Management API references" are. Is this another topic?

// end::deleteLinkedProfilesAdmin[]

// tag::requireProfileLinkingForAdmins[]
[[require-profile-linking]]
= Require Users to Link Anypoint Platform Profiles
// DO NOT USE THIS SECTION YET - NOT YET SUPPORTED 

If your organization has xref:external-identity-index.adoc#enable-profile-linking[enabled the *Link Multiple Accounts* setting], organization administrators can require all users to link available Anypoint Platform profiles.

include::partial$include-nav-steps-org-admin.adoc[]

include::partial$include-nav-identity-providers.adoc[]

. In the *Link Multiple Accounts* section, click *Required*. 
+

> Users can no longer skip the prompt to link their user profiles when logging in to Anypoint Platform.

Reviewer 2: Do you need the last sentence? If so, maybe combine the two sentences: 
...click *Required* so that users must link...

Author: ```suggestion
```

Author: ```suggestion
```

> . In the *Link Multiple Accounts* section, click *Required*. 
> +

Author: ```suggestion
. In the *Link Multiple Accounts* section, click *Required* so that users can't skip the prompt to link their user profiles when logging in to Anypoint Platform. 
```

> . In the *Link Multiple Accounts* section, click *Required*. 
> +

Author: ```suggestion
. In the *Link Multiple Accounts* section, click *Required* so that users must link their user profiles when logging in to Anypoint Platform. 

```

// tag::profileLinkingContextForUsers[]
[[profile-linking-context-users]]
= Linking Your Anypoint Platform Profiles

When you log in to an Anypoint Platform using an email address that is associated with more than one profile in the same organization, you are prompted to link your profiles together. Your options for linking profiles vary based on whether the profiles use the Anypoint identity provider (IdP) or an external IdP that uses single sign-on (SSO). Anypoint Platform supports:

* Linking an SSO profile to a local profile (login managed by Anypoint IdP)
* Linking an SSO profile to another SSO profile that uses a different IdP to log in
* Linking an SSO profile to multiple other profiles that each use different IdPs to log in

Anypoint Platform does not support linking multiple profiles that use the same IdP. You can link up to 25 user profiles.

// end::profileLinkingContextForUsers[]

// tag::profileLinkingForSSOtoLocal[]
[[sso-to-local]]

> = Linking an SSO profile to a Local profile

Author: ```suggestion
= Linking an SSO Profile to a Local Profile
```

When you attempt to log in using SSO and there is another user that shares the same email address but uses Anypoint Platform credentials to log in, you are prompted to link the profiles.

. In the Anypoint Platform login page, click *Sign in with <your identity provider>*.
. If prompted, enter your credentials for your IdP. +
If your SSO credentials share an email address with an profile that is managed by the Anypoint IdP, you are prompted to log in. If you are not prompted, your local profile uses a different email address than your SSO profile.
+
. In the *Link your Account* page, enter your credentials for your local Anypoint Platform profile.
. Click *Link Accounts*.
+
You can now view your linked profiles in the *Identity Providers* section of your user profile.

// end::profileLinkingForSSOtoLocal[]

// tag::profileLinkingSSOtoSSO[]
[[sso-to-sso]]

> = Linking an existing SSO profile to Another SSO profile

Author: ```suggestion
= Linking an Existing SSO Profile to Another SSO Profile
```

[[linking-multiple-profiles]]

When you attempt to login using SSO and there are multiple user profiles that share the same email address but use different identity providers to log in, you are prompted to link the profiles. Note that you cannot link profiles that use the same IdP.

. In the Anypoint Platform login page, click *Sign in with <your identity provider>*.
. If prompted, enter your credentials for your IdP. +
If your SSO credentials share an email address with additional profiles in the same organization, you are prompted to select one of the profiles and log in with those credentials to link your profiles. If you are not prompted, your other profiles use a different email address than the SSO profile you are logging in to.
+
. In the *Link your Account* page, select the profile you want to link. +
You can hover over each profile to see which identity provider it uses.
+
. Click *Link Accounts*.
. Enter your credentials for the profile you chose.
+
You can now view your linked profiles in the *Identity Providers* section of your user profile.

// end::profileLinkingMultiple[]

// tag::profileLinkingContextForAdmins[]
[[enable-profile-linking]]

> = Enabling Users to Link Anypoint Platform profiles

Author: ```suggestion
= Enabling Users to Link Anypoint Platform Profiles
```

// tag::profileLinkingContextForUsers[]
[[profile-linking-context-users]]
= Linking Your Anypoint Platform Profiles

> When you log in to an Anypoint Platform using an email address that is associated with more than one profile in the same organization, you are prompted to link your profiles together. Your options for linking profiles vary based on whether the profiles use the Anypoint identity provider (IdP) or an external IdP that uses single sign-on (SSO). Anypoint Platform supports:

Reviewer: Let me think hard about this : "When you log in to an Anypoint Platform using an email address that is associated with more than one profile in the same organization" 

The user is not actually using an email address to login. The users are logging in from an Idp (i.e. OpenId Idp at this release). The end users may not be immediately aware of which emails they have in their Idp account. Let me take an example. If the Idp requires username (not email) to sign in, the end users type username/password, get redirected to our linking page and see the matched accounts. In this process the email is not displayed. 

What's actually happening is that: (this is for the context; I'm not sure how we put it in here) 
1. the user signs into the IDP 
2. IDP sends to Anypoint a profile (username, email, ... stored in IDP)
3. we use that email to find potential matches. 

So I would say something like "When you log into Anypoint via an identity provider for the first time, we'll find potential user accounts matched by your email address in the Idp profile. You'll be prompted to choose to link your current idp profile to an existing Anypoint account... "

Author: ```suggestion
When you log in to Anypoint Platform for the first time using an external identity provider (IdP), Anypoint Platform locates potential user accounts matched by your email address that is associated with the IdP profile. You are prompted to link your IdP profile with the existing accounts. Your options for linking profiles vary based on whether the profiles use the Anypoint identity provider (IdP) or an external IdP that uses single sign-on (SSO). Anypoint Platform supports:
```

> * Linking an SSO profile to a local profile (login managed by Anypoint IdP)
> * Linking an SSO profile to another SSO profile that uses a different IdP to log in
> * Linking an SSO profile to multiple other profiles that each use different IdPs to log in

Author: ```suggestion
* Linking an SSO login to a local profile by using credentials (login managed by Anypoint IdP)
* Linking an SSO login to another SSO login that uses a different IdP to log in
* Linking an SSO login to multiple other profiles that each use different IdPs to log in
```

> Anypoint Platform does not support linking multiple profiles that use the same IdP. You can link up to 25 user profiles.

Reviewer: Another limitation is that we're supporting Openid providers at this phase, which means all the above Idps are openid Idps.

Author: ```suggestion
Anypoint Platform does not support linking multiple profiles that use the same IdP. The IdP must be an OpenID Connect (OIDC) provider. You can link up to 25 user profiles. This feature is available only if an organization administrator has enabled it.
```

// end::profileLinkingContextForUsers[]

// tag::profileLinkingForSSOtoLocal[]
[[sso-to-local]]
= Linking an SSO Profile to a Local Profile

When you attempt to log in using SSO and there is another user that shares the same email address but uses Anypoint Platform credentials to log in, you are prompted to link the profiles.

. In the Anypoint Platform login page, click *Sign in with <your identity provider>*.
. If prompted, enter your credentials for your IdP. +
If your SSO credentials share an email address with an profile that is managed by the Anypoint IdP, you are prompted to log in. If you are not prompted, your local profile uses a different email address than your SSO profile.
+
. In the *Link your Account* page, enter your credentials for your local Anypoint Platform profile.
. Click *Link Accounts*.
+
You can now view your linked profiles in the *Identity Providers* section of your user profile.

// end::profileLinkingForSSOtoLocal[]

// tag::profileLinkingSSOtoSSO[]
[[sso-to-sso]]

> = Linking an Existing SSO Profile to Another SSO Profile

Reviewer: Can we change the template from "linking profile to profile" to "linking profile to account". Our account model is like one account will have multiple IdP profiles. Well, it also makes sense to say "linking a profile to one / other profiles" . Let me think again

Reviewer: The feature is mainly about linking a SSO profile to an existing account. Functionally, we can't actively link something existing to something else; rather, we link something new to something existing. I like the first section name "linking .. to a local profile" (I prefer local account here). 

What about following the same pattern for the other 2 section title? Say, "Linking a SSO profile to existing sso account"

Author: ```suggestion
= Linking an Existing SSO Login to Another SSO Login
```

+
. Click *Link Accounts*.
. Enter your credentials for the profile you chose.
+
You can now view your linked profiles in the *Identity Providers* section of your user profile.

// end::profileLinkingMultiple[]

// tag::profileLinkingContextForAdmins[]
[[enable-profile-linking]]
= Enabling Users to Link Anypoint Platform Profiles

// DO NOT USE THIS SECTION YET - NOT YET SUPPORTED 

As an organization administrator, you can require users to link multiple Anypoint Platform profiles that use the same email address to log in. 

Anypoint Platform supports:

* Linking an SSO profile to a local profile (profile managed by Anypoint IdP)
* Linking an SSO profile to another SSO profile that uses a different IdP
* Linking an SSO profile to multiple other profiles that use different IdPs

Anypoint Platform does not support linking multiple profiles that use the same IdP.

include::partial$include-nav-steps-org-admin.adoc[]

include::partial$include-nav-identity-providers.adoc[]

. In the *Link Multiple Accounts* section, click *Required*. 
+
Users can no longer skip the prompt to link their user profiles when logging in to Anypoint Platform.

> // end::profileLinkingContextForAdmins[]

Author: ```suggestion
// end::profileLinkingContextForAdmins[]

```

When you attempt to login using SSO and there are multiple user profiles that share the same email address but use different identity providers to log in, you are prompted to link the profiles. Note that you cannot link profiles that use the same IdP.

. In the Anypoint Platform login page, click *Sign in with <your identity provider>*.
. If prompted, enter your credentials for your IdP. +
If your SSO credentials share an email address with additional profiles in the same organization, you are prompted to select one of the profiles and log in with those credentials to link your profiles. If you are not prompted, your other profiles use a different email address than the SSO profile you are logging in to.
+
. In the *Link your Account* page, select the profile you want to link. +
You can hover over each profile to see which identity provider it uses.
+
. Click *Link Accounts*.
. Enter your credentials for the profile you chose.
+
You can now view your linked profiles in the *Identity Providers* section of your user profile.

// end::profileLinkingMultiple[]

// tag::profileLinkingContextForAdmins[]
[[enable-profile-linking]]
= Enabling Users to Link Anypoint Platform Profiles

> // DO NOT USE THIS SECTION YET - NOT YET SUPPORTED 
> 
> As an organization administrator, you can require users to link multiple Anypoint Platform profiles that use the same email address to log in. 
> 
> Anypoint Platform supports:
> 
> * Linking an SSO profile to a local profile (profile managed by Anypoint IdP)
> * Linking an SSO profile to another SSO profile that uses a different IdP
> * Linking an SSO profile to multiple other profiles that use different IdPs
> 
> Anypoint Platform does not support linking multiple profiles that use the same IdP.
> 
> include::partial$include-nav-steps-org-admin.adoc[]
> 
> include::partial$include-nav-identity-providers.adoc[]
> 
> . In the *Link Multiple Accounts* section, click *Required*. 
> +
> Users can no longer skip the prompt to link their user profiles when logging in to Anypoint Platform.

Author: ```suggestion
As an organization administrator, you can enable users to link multiple Anypoint Platform profiles that use the same email address to log in. 

Anypoint Platform supports:

* Linking an SSO login to a local profile by using credentials (login managed by Anypoint IdP)
* Linking an SSO login to another SSO profile that uses a different IdP
* Linking an SSO login to multiple other profiles that use different IdPs

Anypoint Platform does not support linking multiple profiles that use the same IdP. Note that only OpenID Connect (OIDC) IdPs are supported.

include::partial$include-nav-steps-org-admin.adoc[]

include::partial$include-nav-identity-providers.adoc[]

. In the *Link Multiple Accounts* section, click *Enabled*. 
. Click *Save Changes*. 
+
Users in your organization can now link their SSO logins with other IdP profiles that share the same email address.
```

// end::profileLinkingForSSOtoLocal[]

// tag::profileLinkingSSOtoSSO[]
[[sso-to-sso]]
= Linking an Existing SSO Profile to Another SSO Profile

When you attempt to log in using SSO and there is another user that shares the same email address but uses a different external identity provider with SSO, you are prompted to link the profiles.

. In the Anypoint Platform login page, click *Sign in with <your identity provider>*.
. If prompted, enter your credentials for your IdP. +
If your SSO credentials share an email address with another SSO profile in the same organization, you are prompted to log in with those credentials to link your profiles. If you are not prompted, your other SSO profile uses a different email address than the SSO profile you are logging in to.
+
. In the *Link your Account* page, enter your credentials for your local Anypoint Platform profile.
. Click *Link Accounts*.
+
You can now view your linked profiles in the *Identity Providers* section of your user profile.

// end::profileLinkingSSOtoSSO[]

// tag::profileLinkingMultiple[]
[[linking-multiple-profiles]]
= Linking an existing SSO Profile to Multiple Profiles

> = Linking an existing SSO Profile to Multiple Profiles
> = Linking an existing SSO Profile to Multiple Profiles
> = Linking an existing SSO Profile to Multiple Profiles

Author: ```suggestion
```
